test/types/CommunityModel.jl

```diff
@@ -0,0 +1,416 @@
+    add_reactions!(
+        m2,
+        [
+            ReactionForward("r3", Dict("C" => -1, "Ce" => 1)),
+            ReactionForward("EX_C", Dict("Ce" => -1)),
+            ReactionBidirectional("EX_A", Dict("Ae" => -1)),
+            ReactionBidirectional("r1", Dict("Ae" => -1, "A" => 1)),
+            ReactionForward("r2", Dict("A" => -1, "C" => 1)), #bm2
+        ],
+    )
+
+    cm1 = CommunityMember(
+        model = m1,
+        exchange_reaction_ids = ["EX_A", "EX_B"],
+        biomass_reaction_id = "r2",
+    )
+    @test contains(sprint(show, MIME("text/plain"), cm1), "community member")
+
+    cm2 = CommunityMember(
+        model = m2,
+        exchange_reaction_ids = ["EX_A", "EX_C"],
+        biomass_reaction_id = "r2",
+    )
+
+    cm = CommunityModel(
+        members = OrderedDict("m1" => cm1, "m2" => cm2),
+        abundances = [0.2, 0.8],
+        environmental_links = [
+            EnvironmentalLink("EX_A", "Ae", -10.0, 10.0)
+            EnvironmentalLink("EX_B", "Be", -20.0, 20.0)
+            EnvironmentalLink("EX_C", "Ce", -30, 30)
+        ],
+    )
+    @test contains(sprint(show, MIME("text/plain"), cm), "community model")
+
+    @test issetequal(
+        variables(cm),
+        [
+            "m1#EX_A"
+            "m1#r1"
+            "m1#r2"
+            "m1#r3"
+            "m1#EX_B"
+            "m2#r3"
+            "m2#EX_C"
+            "m2#EX_A"
+            "m2#r1"
+            "m2#r2"
+            "EX_A"
+            "EX_C"
+            "EX_B"
+        ],
+    )
+
+    @test issetequal(
+        metabolites(cm),
+        [
+            "m1#A"
+            "m1#B"
+            "m1#Ae"
+            "m1#Be"
+            "m2#Ae"
+            "m2#A"
+            "m2#C"
+            "m2#Ce"
+            "ENV_Ae"
+            "ENV_Be"
+            "ENV_Ce"
+        ],
+    )
+
+    @test issetequal(
+        genes(cm),
+        [
+            "m1#g1"
+            "m1#g2"
+            "m1#g3"
+            "m1#g4"
+            "m2#g1"
+            "m2#g2"
+            "m2#g3"
+            "m2#g4"
+        ],
+    )
+
+    @test n_variables(cm) == 13
+    @test n_metabolites(cm) == 11
+    @test n_genes(cm) == 8
+
+    @test all(
+        stoichiometry(cm) .== [
+            0.0 1.0 -1.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
+            0.0 0.0 1.0 -1.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
+            -1.0 -1.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
+            0.0 0.0 0.0 1.0 -1.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0
+            0.0 0.0 0.0 0.0 0.0 0.0 0.0 -1.0 -1.0 0.0 0.0 0.0 0.0
+            0.0 0.0 0.0 0.0 0.0 0.0 0.0 0.0 1.0 -1.0 0.0 0.0 0.0
+            0.0 0.0 0.0 0.0 0.0 -1.0 0.0 0.0 0.0 1.0 0.0 0.0 0.0
+            0.0 0.0 0.0 0.0 0.0 1.0 -1.0 0.0 0.0 0.0 0.0 0.0 0.0
+            0.2 0.0 0.0 0.0 0.0 0.0 0.0 0.8 0.0 0.0 -1.0 0.0 0.0
+            0.0 0.0 0.0 0.0 0.2 0.0 0.0 0.0 0.0 0.0 0.0 -0.2 0.0
+            0.0 0.0 0.0 0.0 0.0 0.0 0.8 0.0 0.0 0.0 0.0 0.0 -0.8
+        ],
+    )
```

Comment:
There are no dicts in the way so this doesn't depend on dict order, right?

Reply:
nope, I purposefully used either vectors or `OrderedDict`s to make this work